File: solutions/Python/2206.DivideArrayIntoEqualPairs.py
from typing import List

class Solution:
    def divideArray(self, nums: List[int]) -> bool:
        if len(nums) < 2 or len(nums) % 2 != 0:
            return False;

        pairs_needed = len(nums) // 2
        pairs = 0
        num_map = {}

        for num in nums:
            num_map[num] = num_map.get(num, 0) + 1

            if num_map[num] % 2 == 0:
                pairs += 1

        return pairs == pairs_needed

File: solutions/TypeScript/2206.DivideArrayIntoEqualPairs.ts
function divideArray(nums: number[]): boolean {
    if (nums.length < 2 || nums.length % 2 !== 0) return false;

    const pairsLength: number = nums.length / 2;
    let pairs: number = 0;
    let numsMap: Map<number, number> = new Map();

    for (const num of nums) {
        if (numsMap.has(num)) {
            let valor: number = numsMap.get(num);
            valor += 1;
            numsMap.set(num, valor);
            if (valor % 2 === 0) {
                pairs++;
            }
        } else {
            numsMap.set(num, 1);
        }
    }
    return pairs === pairsLength;
};